### DISCUSSION TITLE
Material UI Checkboxes with FormGroup not returning expected values

### DISCUSSION THREAD

**Discussion by aadlc** [react-hook-form/react-hook-form#4920] [General] [answered]
In the previous version, I was able to create a checkbox group as follow:
```
<FormControl
    variant="outlined"
    margin="normal"
    component="fieldset"
  >
      <FormLabel component="legend">
        Please select the category:
      </FormLabel>
      <FormGroup>
        <FormControlLabel
          control={<Checkbox name="checkbox" inputRef={register} />}
          value="Option1"
          label="Option 1"
        />
        <FormControlLabel
          control={<Checkbox name="checkbox" inputRef={register} />}
          value="Option2"
          label="Option 2"
        />
     </FormGroup>>
</FormControl>
```
And the checkbox output, if both checked, would be:
`{"checkbox":["Option1","Option2"]}`
 
Working example here https://codesandbox.io/s/mui-rfhv6-3yye6

On version 7, I am having a hard time achieving the same:

```
<FormControl
    variant="outlined"
    margin="normal"
    component="fieldset"
    fullWidth
  >
    <FormGroup>
      <FormControlLabel
        control={
          <Controller
            name="Checkbox"
            control={control}
            render={({ field }) => <Checkbox {...field} />}
          />
        }
        value="Option1"
        label="Option 1"
      />
      <FormControlLabel
        control={
          <Controller
            name="Checkbox"
            control={control}
            render={({ field }) => <Checkbox {...field} />}
          />
        }
        value="Option2"
        label="Option 2"
      />
    </FormGroup>
</FormControl>
```
And its output is:
`{"Checkbox":true}`
Working example here https://codesandbox.io/s/mui-rfhv7-hb362

Any idea what am I missing?

**Comment by bluebill1049** [comment, marked as answer]
First of all, for controlled input most likely is safer to use `Controller` or `useController`, because some internal component may have an animation which triggers by local state update, but for checkbox we are good. 

**Register**

here is a working version with v7: 

https://codesandbox.io/s/mui-rfhv6-forked-5yn02?file=/src/App.js

```
const { ref, ...rest } = register("checkbox");
```

The important part is to assign ref correctly, because by default register doesn't return inputRef, so we have assigned that manually.

**Controller**

for `Controller`, you are supposed to group all your controlled `checkedbox` and let a single Controller collect value and do validation.

**Reply by aadlc** [reply]
Thank you for the explanation and the code!

**Reply by simplenotezy** [reply]
I have a similar issue, described here: https://github.com/react-hook-form/react-hook-form/discussions/5762 I have tried implementing your suggestion, but I still only get boolean values returned. Not sure what's wrong here.

**Reply by sonphung205** [reply]
@simplenotezy does your problem solved? I have the same issue with my native input.
